### PR TITLE
added basic string manipulation functions

### PR DESCRIPTION
**Pull request by brennanjl**
Added basic string manipulation functions. These are taken from the top-level SQL functions identified in Postgres: https://www.postgresql.org/docs/16/functions-string.html

This was requested by Truflation here: https://github.com/truflation/tsn/issues/284#issuecomment-2136359976

Documentation for supported functions can be found here: https://github.com/kwilteam/docs/pull/105/files#diff-aacab0e9719a83529316f1b99149bb81e25da9b918b53725d801890376d811a7

@outerlook does this cover what you need?

### CODE REVIEW

**Review by jchappelow — APPROVED** [review]
The list of functions looks fine to me.   Scanned the checks quickly and they also look fine.  Only question is if we need/want to support all the synonyms like char_length/character_length.  The tech burden isn't bad.

**Comment by brennanjl** [comment]
Yeah I wasn't sure. I'd be open to removing one of them, but also have no strong preference.

**Review by Yaiba — DISMISSED** [review]
LGTM

File: internal/engine/integration/procedure_test.go [hunk 1]
```diff
@@ -160,6 +160,60 @@ func Test_Procedures(t *testing.T) {
...
+				if $val != 'hello world!' {
+					error('rpad failed');
+				}
+				if overlay($val, 'xx', 2, 5) != 'hxxworld!' {
+					error('overlay failed');
+				}
+				if position('world', $val) != 7 {
+					error('position failed');
+				}
+				if substring($val, 7, 5) != 'world' {
```
Comment (Yaiba) [inline internal/engine/integration/procedure_test.go:208]: maybe also add test for different args? functions like `ltrim`,`overlay`,`substring`(and more) support different args num

Comment (brennanjl) [reply internal/engine/integration/procedure_test.go:228]: ok done

**Review by Yaiba — APPROVED** [review]

**Pull request merged**